File: output_CodeWhisperer/java/Easy/75-reverse-vowels-of-a-string.java
/**https://leetcode.com/problems/reverse-vowels-of-a-string/ */
//Given a string s, reverse only all the vowels in the string and return it.
//The vowels are 'a', 'e', 'i', 'o', and 'u', and they can appear in both lower and upper cases, more than once.
// 
//Example 1:
//Input: s = "IceCreAm"
//Output: "AceCreIm"
//Explanation:
//The vowels in s are ['I', 'e', 'e', 'A']. On reversing the vowels, s becomes "AceCreIm".
//Example 2:
//Input: s = "leetcode"
//Output: "leotcede"
// 
//Constraints:
//	1 <= s.length <= 3 * 105
//	s consist of printable ASCII characters.
class Solution {
    public String reverseVowels(String s)
    {
        char[] str = s.toCharArray();
        int i = 0;
        int j = str.length - 1;
        while (i < j)
        {
            while (i < j && !isVowel(str[i]))
                i++;
            while (i < j && !isVowel(str[j]))
                j--;
            char temp = str[i];
            str[i] = str[j];
            str[j] = temp;
            i++;
            j--;
        }
        return new String(str);
        
    }
}

File: Output/output_TY/c/Easy/75-reverse-vowels-of-a-string.c
/**https://leetcode.com/problems/reverse-vowels-of-a-string/ */
//Given a string s, reverse only all the vowels in the string and return it.
//The vowels are 'a', 'e', 'i', 'o', and 'u', and they can appear in both lower and upper cases, more than once.
// 
//Example 1:
//Input: s = "hello"
//Output: "holle"
//Example 2:
//Input: s = "leetcode"
//Output: "leotcede"
// 
//Constraints:
//	1 <= s.length <= 3 * 105
//	s consist of printable ASCII characters.
char * reverseVowels(char * s){
    char *p = s;
    char *q = s + strlen(s) - 1;
    char *vowels = "aeiouAEIOU";
    while (p < q) {
        while (*p != *vowels && p < q) p++;
        while (*q != *vowels && p < q) q--;
        if (p < q) {
            char tmp = *p;
            *p++ = *q;
        }
    }
    return s;
}